src/utilities/loading-utils.ts

```diff
@@ -4,10 +4,17 @@ const sessionStorageStartTimeItem = 'loading-time-start';
 const sessionStorageMessageItem = 'loading-message';
 const messageSeparator = '<br/>';
 
+//loadingMeasurements is a global object (declared in index.html)
+//keys are action start and end labels, values are milliseconds
+declare global {
+  interface Window { loadingMeasurements: Record<string, number> }
+}
+const loadingMeasurements = window.loadingMeasurements;
+
 function getTimestamp() {
   const start = sessionStorage.getItem(sessionStorageStartTimeItem);
   if (!start) {
-    return "??";
```

Comment:
I'm suggesting this get changed to `null` for cleaner type checking in `showLoadingMessage` and `removeLoadingMessage`